api/scm.go

```diff
@@ -191,7 +197,41 @@ func SyncRepo(c *gin.Context) {
 
 			return
 		}
+
+		err = updateUserFavorites(c, r)
+		if err != nil {
+			util.HandleError(c, http.StatusInternalServerError, err)
+			return
+		}
 	}
 
 	c.JSON(http.StatusOK, fmt.Sprintf("repo %s synced", r.GetFullName()))
 }
+
+// updateUserFavorites is a helper function that takes a repo
+// that has been deleted from the SCM and ensures that it does
+// not appear in the user's favorites after its deletion.
+func updateUserFavorites(c *gin.Context, r *library.Repo) error {
```

Comment:
@ecrupper the change looks good to me so I'll approve 👍 

One point of clarification I wanted to inquire about for my own understanding.

This change doesn't update `Favorites` for all users in Vela, right?

AFAICT, it appears it would update `Favorites` for the owner of the repo in Vela.

So any users who had the repo in their `Favorites`, but wasn't the owner of that repo, would still see it?